pipeline/pipeline.go

```diff
@@ -487,24 +552,31 @@ func (p *Pipeline) maintenance() {
 		p.antispamer.maintenance()
 		p.metricsHolder.maintenance()
 
-		totalCommitted := p.totalCommitted.Load()
-		deltaCommitted := int(totalCommitted - lastCommitted)
-
-		totalSize := p.totalSize.Load()
-		deltaSize := int(totalSize - lastSize)
+		// set new values and get the diffs
+		deltaInputEvents := inputEvents.setAndGetDiff(p.inputEvents.Load())
+		deltaInputSize := inputSize.setAndGetDiff(p.inputSize.Load())
+		deltaOutputEvents := outputEvents.setAndGetDiff(p.outputEvents.Load())
+		deltaOutputSize := outputSize.setAndGetDiff(p.outputSize.Load())
 
-		rate := int(float64(deltaCommitted) * float64(time.Second) / float64(interval))
-		rateMb := float64(deltaSize) * float64(time.Second) / float64(interval) / 1024 / 1024
-
-		tc := totalCommitted
-		if totalCommitted == 0 {
-			tc = 1
+		// increment the metrics
+		getCounter := func(metric string) prometheus.Counter {
```

Comment:
Is it required?
`stats.GetCounter(p.subsystemName(), metric).Add(float64(deltaOutputSize))` seems good.
And can be simplified to avoid `float64(deltaOutputSize)`.

Reply:
Was thinking the same thing, I would prefer straight calls

Reply:
done, thank you